### PR TITLE
feat: initial checkin of custom fly-by-wire code

### PR DESCRIPTION
**Pull request by Benjozork**
## Summary of Changes

* Initial check-in of custom FBW code
* **NOTE: THIS IS NOT ENABLED BY DEFAULT !**

Discord username (if different from GitHub): someperson#4953

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by ImenesFBW** [comment]
For anyone downloading this PR, it is still in the testing phase, use at your own risk, don't expect support on discord for it.

**Comment by Snapmatics** [comment]
This is a preview integration of a custom FBW with A32NX.

⚠️ As of today, the custom FBW is not compatible with YourControls plugin.

⚠️ Interaction with ATHR (autothrottle) has major issues.
> Throttle Levers are completely locked as soon as ATHR is enabled.
> You need to manually disable ATHR to be able to move them again

File: src/fbw/build.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  ./src/model/FlyByWire_data.cpp \
+  -I ./src/interface \
+  ./src/interface/SimConnectInterface.cpp \
+  ./src/FlyByWireInterface.cpp \
+  ./src/main.cpp \
+  -o ./build/airbus-fly-by-wire.wasm
+
+wasm-opt -O4 --strip-debug --strip-producers ./build/airbus-fly-by-wire.wasm -o ./build/airbus-fly-by-wire.wasm
+
+cp ./build/airbus-fly-by-wire.wasm ../../A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/panel/airbus-fly-by-wire.wasm
```
Comment (devsnek) [inline src/fbw/build.sh:34]:
```
OUTPUT=../../A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/panel/airbus-fly-by-wire.wasm
```
```suggestion
  -o $OUTPUT

wasm-opt -O4 --strip-debug --strip-producers $OUTPUT -o $OUTPUT
```

File: src/fbw/build.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -ex
+
+mkdir -p ./build
```
Comment (devsnek) [inline src/fbw/build.sh:5]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -6,8 +6,9 @@
     "lint-fix": "npm run lint -- --fix",
     "prettier": "prettier --write **/*.json **/*.yml src/instruments/**/*.css",
     "build:instruments": "rollup -c src/instruments/rollup.config.js",
+    "build:fbw": "cd src/fbw && ./build.sh",
```
Comment (devsnek) [inline package.json:9]:
```suggestion
```

File: package.json [hunk 4]
```diff
@@ -6,8 +6,9 @@
...
     "build:msfs": "node scripts/build.js",
-    "build": "npm run build:instruments && npm run build:msfs",
+    "build": "npm run build:instruments && npm run build:fbw && npm run build:msfs",
```
Comment (devsnek) [inline package.json:11]:
```suggestion
    "build": "npm run build:instruments && npm run build:msfs",
```

and then add `../src/fbw/build.sh` to `scripts/build.sh`

**Comment by ErikNL80** [comment]
I don't know if I can add my test results here? Otherwise, please delete! 
Anyway downloaded "built": "2020-11-29T18:43:27+00:00" and did a flight from KSBA to KSJC. My overall findings:

1. I manually deleted the a32nx folder from %LOCALAPPDATA%\Packages\Microsoft.FlightSimulator_8wekyb3d8bbwe, the system does not generate a new one, and also the throttle config file is not created there. Could not test these values with my TCA throttle. Also using the downloader first would not generate this folder, although it was there somehow?
2. Flight went well. A/P behaved, airplane followed flightplan with a slight off-set to the left
3. Rocking plane on cruise (FL240). Managed by disabling SEC2 (overhead right)
4. ILS landing, also an off-set in following the loc, but maybe 2,3 and 4 need to be solved via other PR.

Many, many thanks for the amount of work and effort you put into this!

**Comment by Benjozork** [comment]
@ErikNL80 sorry, there was a miscommunication here and the code present in this PR does not include the throttle configs. that will be added in a future PR.

**Comment by ErikNL80** [comment]
No problems!

**Review by kberg93 — APPROVED** [review]
Seems fine, nothing wanted to kill me 👍🏻

**Pull request merged**